src/cxx.cc

```diff
@@ -132,6 +136,34 @@ String::String(const char16_t *s, std::size_t len) {
              len);
 }
 
+String String::lossy(const std::string &s) noexcept {
+  return String::lossy(s.data(), s.length());
+}
+
+String String::lossy(const char *s) noexcept {
+  assert(s != nullptr);
+  return String::lossy(s, std::strlen(s));
+}
+
+String String::lossy(const char *s, std::size_t len) noexcept {
+  assert(s != nullptr || len == 0);
+  String string;
+  cxxbridge1$string$from_utf8_lossy(&string, s, len);
```

Comment:
`from_utf8_lossy` just passes this ptr to `slice::from_raw_parts`, which is UB if the pointer is null.

https://github.com/dtolnay/cxx/blob/316f565f6954bf339f247b3e5f8d345847bc52c8/src/symbols/rust_string.rs#L42-L48

This code needs to substitute in some non-null pointer instead, like happens here:

https://github.com/dtolnay/cxx/blob/c104b0a9d55aa0e709146b3633f2729733adeadc/src/cxx.cc#L118

Reply:
Huh! TIL. Adjusted accordingly.